test/testkit/k8s/deployment.go

```diff
@@ -8,28 +8,48 @@ import (
 )
 
 type Deployment struct {
-	name      string
-	namespace string
-	replicas  int32
-	labels    map[string]string
-	podSpec   corev1.PodSpec
+	name        string
+	namespace   string
+	replicas    int32
+	labels      map[string]string
+	annotations map[string]string
+	podSpec     corev1.PodSpec
 }
 
 func NewDeployment(name, namespace string) *Deployment {
 	return &Deployment{
-		name:      name,
-		namespace: namespace,
-		replicas:  1,
-		labels:    make(map[string]string),
-		podSpec:   SleeperPodSpec(),
+		name:        name,
+		namespace:   namespace,
+		replicas:    1,
+		labels:      make(map[string]string),
+		annotations: make(map[string]string),
+		podSpec:     SleeperPodSpec(),
 	}
 }
 
+func (d *Deployment) WithName(name string) *Deployment {
```

Comment:
NewDeployment already takes a name as ctor argument, since it's not really optional. Do we need an WithXxx function?